### ISSUE TITLE
Get Filtered Rows

### ISSUE THREAD

**Issue by Rumeth** [angular-ui/ui-grid#4583]
UI Grid #3.0.6

The getVisibleRows(grid) returns all the rows visible in the Grid.

However, if the Grid is paginated, only the rows visible in the current page are being returned.

Is there an option to get all the filtered rows from the Grid?

**Comment by thynctank** [comment]
I don't believe there is anything for this yet, which is an issue for me at the moment as there are situations where sorting or overwriting gridOptions.data can cause gridOptions.totalItems and the other item counts to be incorrect, so I'm looking for the right way to handle that myself.

**Comment by greggriffin** [comment]
I need a solution for this as well.

**Comment by crazyjat** [comment]
This is a very important missing feature

**Comment by Raystorm7** [comment]
Please implement a solution for this missing feature :)

**Comment by patriciocordova** [comment]
+1

**Comment by raky291** [comment]
+1

**Comment by nikulsuthar** [comment]
+1

**Comment by HaythemJ** [comment]
+1

**Comment by umdstu** [comment]
I'm experiencing the opposite behavior. When pagination is on, and I navigate to page 2, getVisibleRows() returns all rows from page 1 & 2. If I navigate to page 3, getVisibleRows() returns all rows from pages 1, 2, & 3. 

What I want to get is what you all seem to be getting, the rows literally visible to the user, which I would expect to be only those rows for the current page.

**Comment by chirag-daraji-pmc** [comment]
+1

**Comment by LordNashor** [comment]
+1
